src/network/network_gui.cpp

```diff
@@ -2174,6 +2176,32 @@ struct NetworkCompanyPasswordWindow : public Window {
 		this->SetFocusedWidget(WID_NCP_PASSWORD);
 	}
 
+	void UpdateWarningStringSize()
+	{
+		assert(this->nested_root->smallest_x > 0);
+		this->warning_size.width = this->nested_root->current_x - (WD_FRAMETEXT_LEFT + WD_FRAMETEXT_RIGHT + WD_FRAMERECT_LEFT + WD_FRAMERECT_RIGHT);
+		this->warning_size.height = GetStringHeight(STR_WARNING_PASSWORD_SECURITY, this->warning_size.width);
+		this->warning_size.height += WD_FRAMETEXT_TOP + WD_FRAMETEXT_BOTTOM + WD_FRAMERECT_TOP + WD_FRAMERECT_BOTTOM;
```

Comment:
Is it useful if we put this "algorithm" in a function? You now use it at two places, but if I understand it correctly, it can also be useful for other places where we want this kind of behaviour? (honest question btw, no is a valid answer :D)

Reply:
I think the best would be to add a flag to label controls or similar, that they have height dependent on width (and width is not measured for the widget), which then causes them to wrap and size after the main size calculations are done.